Package does not build with >=zeek-7.1

This package still uses and implements `OpaqueVal::DoSerialize` and `OpaqueVal::DoUnserialize` which were deprecated since 7.0 and are removed with 7.1,

```
[  0%] Built target Zeek_MoreHashes_symlink
[ 11%] Building CXX object CMakeFiles/Zeek_MoreHashes.dir/morehashes.bif.cc.o
[ 22%] Building CXX object CMakeFiles/Zeek_MoreHashes.dir/src/Plugin.cc.o
[ 33%] Building CXX object CMakeFiles/Zeek_MoreHashes.dir/src/MMH3.cc.o
In file included from /Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.cc:1:
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:26:54: error: only virtual member functions can be marked 'override'
   26 |   broker::expected<broker::data> DoSerialize() const override;
      |                                                      ^~~~~~~~
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:28:48: error: only virtual member functions can be marked 'override'
   28 |   bool DoUnserialize(const broker::data &data) override;
      |                                                ^~~~~~~~
In file included from /Users/bbannier/src/packages/zeek/zeek-more-hashes/src/Plugin.cc:3:
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:26:54: error: only virtual member functions can be marked 'override'
   26 |   broker::expected<broker::data> DoSerialize() const override;
      |                                                      ^~~~~~~~
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:28:48: error: only virtual member functions can be marked 'override'
   28 |   bool DoUnserialize(const broker::data &data) override;
      |                                                ^~~~~~~~
In file included from /Users/bbannier/src/packages/zeek/zeek-more-hashes/src/morehashes.bif:2:
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:26:54: error: only virtual member functions can be marked 'override'
   26 |   broker::expected<broker::data> DoSerialize() const override;
      |                                                      ^~~~~~~~
/Users/bbannier/src/packages/zeek/zeek-more-hashes/src/MMH3.h:28:48: error: only virtual member functions can be marked 'override'
   28 |   bool DoUnserialize(const broker::data &data) override;
      |                                                ^~~~~~~~
2 errors generated.
2 errors generated.
make[2]: *** [CMakeFiles/Zeek_MoreHashes.dir/src/Plugin.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Zeek_MoreHashes.dir/morehashes.bif.cc.o] Error 1
2 errors generated.
make[2]: *** [CMakeFiles/Zeek_MoreHashes.dir/src/MMH3.cc.o] Error 1
make[1]: *** [CMakeFiles/Zeek_MoreHashes.dir/all] Error 2
make: *** [all] Error 2
```

--- a/.github/workflows/zeek-matrix.yml
+++ b/.github/workflows/zeek-matrix.yml
@@ -14,11 +14,11 @@ jobs:
       matrix:
         zeekver: [zeek, zeek-lts, zeek-nightly]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: zeek/action-zkg-install@v2
         with:
           zeek_version: ${{ matrix.zeekver }}
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         if: failure()
         with:
           name: zkg-logs-${{ matrix.zeekver }}

--- a/.github/workflows/zeek-nightly.yml
+++ b/.github/workflows/zeek-nightly.yml
@@ -12,7 +12,7 @@ jobs:
         with:
           pkg: ${{ github.server_url }}/${{ github.repository }}
           zeek_version: 'zeek-nightly'
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         if: failure()
         with:
           name: zkg-logs

--- a/src/MMH3.cc
+++ b/src/MMH3.cc
@@ -21,47 +21,40 @@ uint32_t detail::MMH3Val::Result() {
   return PMurHash32_Result(mmh3_h1, mmh3_carry, mmh3_total_len);
 }
 
-broker::expected<broker::data> detail::MMH3Val::DoSerialize() const {
+std::optional<zeek::BrokerData> detail::MMH3Val::DoSerializeData() const {
   if (!IsValid())
-    return {broker::vector{false}};
+    return std::nullopt;
 
-  broker::vector d = {true, mmh3_h1, mmh3_carry, mmh3_total_len};
-  return {std::move(d)};
+  zeek::BrokerListBuilder builder;
+  builder.Reserve(4);
+  builder.Add(true);
+  builder.Add(static_cast<uint64_t>(mmh3_h1));
+  builder.Add(static_cast<uint64_t>(mmh3_carry));
+  builder.Add(static_cast<uint64_t>(mmh3_total_len));
+
+  return std::move(builder).Build();
 }
 
 // This is a totally untested best-effort implementation. It's a
 // bit of a burden adding this to a HashVal.
-bool detail::MMH3Val::DoUnserialize(const broker::data &data) {
-  auto *d = broker::get_if<broker::vector>(data);
-  if (!d)
+bool detail::MMH3Val::DoUnserializeData(zeek::BrokerDataView data) {
+  if (!data.IsList())
     return false;
 
-  const auto &dv = *d;
+  auto d = data.ToList();
 
-  if (dv.size() < 1)
+  if (d.Size() != 4)
     return false;
 
-  auto *valid = broker::get_if<broker::boolean>(dv[0]);
-  if (!valid)
+  if (!d[0].IsBool() || !d[1].IsCount() || !d[2].IsCount() || !d[3].IsCount())
     return false;
 
-  if (!*valid)
-    return true; // while the content is invalid, unserializing was
-                 // successful.
+  if (!d[0].ToBool()) // This shouldn't actually happen.
+    return true;
 
-  if (dv.size() < 4)
-    return false;
-
-  auto *d_mmh3_h1 = broker::get_if<broker::count>(dv[1]);
-  auto *d_mmh3_carry = broker::get_if<broker::count>(dv[2]);
-  auto *d_mmh3_total_len = broker::get_if<broker::count>(dv[3]);
-
-  if (!d_mmh3_h1 || !d_mmh3_carry || !d_mmh3_total_len)
-    return false;
-
-  mmh3_h1 = static_cast<uint32_t>(*d_mmh3_h1);
-  mmh3_carry = static_cast<uint32_t>(*d_mmh3_carry);
-  mmh3_total_len = static_cast<uint32_t>(*d_mmh3_total_len);
+  mmh3_h1 = static_cast<uint32_t>(d[1].ToCount());
+  mmh3_carry = static_cast<uint32_t>(d[2].ToCount());
+  mmh3_total_len = static_cast<uint32_t>(d[3].ToCount());
 
   return true;
 }

--- a/src/MMH3.h
+++ b/src/MMH3.h
@@ -4,6 +4,7 @@
 #include <broker/error.hh>
 
 #include <zeek/OpaqueVal.h>
+#include <zeek/broker/Data.h>
 #include <zeek/file_analysis/analyzer/hash/Hash.h>
 #include <zeek/file_analysis/analyzer/hash/events.bif.h>
 
@@ -23,9 +24,9 @@ public:
 
   uint32_t Result();
 
-  broker::expected<broker::data> DoSerialize() const override;
+  std::optional<zeek::BrokerData> DoSerializeData() const override;
 
-  bool DoUnserialize(const broker::data &data) override;
+  bool DoUnserializeData(zeek::BrokerDataView data) override;
 
   const char *OpaqueName() const override { return "MMH3Val"; }
 

--- a/tests/mmh3/bif.zeek
+++ b/tests/mmh3/bif.zeek
@@ -1,19 +1,11 @@
 # @TEST-EXEC: zeek -b Zeek::MoreHashes %INPUT >out
 # @TEST-EXEC: btest-diff out
 
-function my_assert(r: bool, msg: string) {
-	if ( r )
-		return;
-
-	print msg;
-	exit(1);
-}
-
 event zeek_init() &priority=0 {
 	local h = mmh3_hash_init();
 	mmh3_hash_update(h, "zeek");
 	local r = mmh3_hash_finish(h);
-	my_assert(r == 1898642774, fmt("expected 1898642774 got %s", r));
+	assert r == 1898642774, fmt("expected 1898642774 got %s", r);
 	print "\"zeek\"", r;
 }
 
@@ -21,7 +13,7 @@ event zeek_init() &priority=-1 {
 	local h = mmh3_hash_init(42);
 	mmh3_hash_update(h, "zeek");
 	local r = mmh3_hash_finish(h);
-	my_assert(r == 162293119, fmt("expected 162293119 got %s", r));
+	assert r == 162293119, fmt("expected 162293119 got %s", r);
 	print "\"zeek\" with seed 42", r;
 }
 
@@ -32,7 +24,7 @@ event zeek_init() &priority=-2 {
 	mmh3_hash_update(h, "e");
 	mmh3_hash_update(h, "k");
 	local r = mmh3_hash_finish(h);
-	my_assert(r == 162293119, fmt("expected 162293119 got %s", r));
+	assert r == 162293119, fmt("expected 162293119 got %s", r);
 	print "\"zeek\" incremental with seed 42", r;
 }
 
@@ -41,18 +33,18 @@ event zeek_init() &priority=-3 {
 	mmh3_hash_update(h, "z");
 	mmh3_hash_update(h, "e");
 	local r = mmh3_hash_finish(h);
-	my_assert(r == 2907484762, fmt("expected 2907484762 got %s", r));
+	assert r == 2907484762, fmt("expected 2907484762 got %s", r);
 	print "\"ze\" incremental", r;
 }
 
 event zeek_init() &priority=-4 {
 	local r = mmh3_hash("zeek");
 	print "\"zeek\" oneshot", r;
-	my_assert(r == 1898642774, fmt("expected 1898642774 got %s", r));
+	assert r == 1898642774, fmt("expected 1898642774 got %s", r);
 }
 
 event zeek_init() &priority=-5 {
 	local r = mmh3_hash("zeek", 42);
 	print "\"zeek\" oneshot with seed 42", r;
-	my_assert(r == 162293119, fmt("expected 162293119 got %s", r));
+	assert r == 162293119, fmt("expected 162293119 got %s", r);
 }

